userdocs/diagnostics/multiple-class-inheritance.md

```diff
@@ -0,0 +1,44 @@
+# Multiple Inheritance of Classes
+
+In some programming languages, a class can inherit the interface of multiple base classes. Known as multiple inheritance, this feature can add significant complexity to the language and is unsupported in Swift. Instead, Swift allows composition of interfaces using protocols.
+
+Consider the following example with a protocol `Utensil` and various conforming types of specific utensils:
+
+```swift
+protocol Utensil { 
+    var name: String {get set}
+} 
+
+
+protocol ServingUtensil: Utensil {
+    func serve()
+} 
+
+extension ServingUtensil {
+    func serve() {
+        print("Serving food to guests...")
+    }
+}
+
+
+protocol Fork: Utensil {
+    func spear()
+}
+
+extension Fork {
+    func spear() { 
+        print("Spearing food...")
+    }
+}
+
+
+struct CarvingFork: ServingUtensil, Fork {
+    var name = "Carving Fork"
+}
```

Comment:
A big part of the didactic value of my suggested example was that `CarvingFork`, `Ladle`, and `Spork` show how protocols can be composed in various intersecting ways. It illustrates the power of protocol composition in (IMO) a motivating way. When you use only a single concrete type, I think you also lose that key message.

This is why I'm encouraging you to trim everything from the example that doesn't serve a direct didactic purpose (like the `print` statements), to make room for the parts of the example that are important for teaching.